sdk/storage/Azure.Storage.Common/src/Shared/StorageClientOptions.cs

```diff
@@ -42,6 +42,23 @@ public static void Initialize(this ClientOptions options)
             new StorageSharedKeyPipelinePolicy(
                 credential ?? throw Errors.ArgumentNull(nameof(credential)));
 
+        /// <summary>
+        /// Get an authentication policy to sign Storage requests.
+        /// </summary>
+        /// <param name="credential">Credential to use.</param>
+        /// <param name="serviceUri">Service Uri. Must not contain shared access signature.</param>
+        /// <returns>An authentication policy.</returns>
+        public static HttpPipelinePolicy AsPolicy(this AzureSasCredential credential, Uri serviceUri)
+        {
+            var queryParameters = serviceUri.GetQueryParameters();
```

Comment:
Do you need to check if `serviceUri` is null?  I think this will get called before validation in `base(...)` for some .ctors.  Consider also taking in the name of the URI param so you can throw the most helpful exception.

Reply:
I could. However, after writing test for that and wrapping this piece of code in null check I found that `NullReferenceException` is thrown [here](https://github.com/Azure/azure-sdk-for-net/blob/1588b45c82b96690d8211b3ef5f9e9d9b095de06/sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs#L357). 
Maybe we should add `Argument.AssertNotNull` in both `AsPolicy` and ctros ?

Reply:
Added validations.